festim/hydrogen_transport_problem.py

```diff
@@ -711,3 +713,506 @@ def post_processing(self):
+                    * dx
+                )
+
+        # add fluxes
+        for bc in self.boundary_conditions:
+            if isinstance(bc, F.ParticleFluxBC):
+                # check that the bc is applied on a surface
+                # belonging to this subdomain
+                if subdomain == self.surface_to_volume[bc.subdomain]:
+                    v = bc.species.subdomain_to_test_function[subdomain]
+                    form -= bc.value_fenics * v * self.ds(bc.subdomain.id)
+
+        # add volumetric sources
+        for source in self.sources:
+            v = source.species.subdomain_to_test_function[subdomain]
+            if source.volume == subdomain:
+                form -= source.value_fenics * v * dx
+
+        # store the form in the subdomain object
+        subdomain.F = form
+
+    def create_formulation(self):
+        """
+        Takes all the formulations for each subdomain and adds the interface conditions.
+
+        Finally compute the jacobian matrix and store it in the ``J`` attribute,
+        adds the ``entity_maps`` to the forms and store them in the ``forms`` attribute
+        """
+        mesh = self.mesh.mesh
+        mt = self.facet_meshtags
+
+        for interface in self.interfaces:
+            interface.mesh = mesh
+            interface.mt = mt
+
+        integral_data = [
+            interface.compute_mapped_interior_facet_data(mesh)
+            for interface in self.interfaces
+        ]
+        [interface.pad_parent_maps() for interface in self.interfaces]
+        dInterface = ufl.Measure("dS", domain=mesh, subdomain_data=integral_data)
+
+        def mixed_term(u, v, n):
+            return ufl.dot(ufl.grad(u), n) * v
+
+        n = ufl.FacetNormal(mesh)
+        cr = ufl.Circumradius(mesh)
+
+        entity_maps = {
+            sd.submesh: sd.parent_to_submesh for sd in self.volume_subdomains
+        }
+        for interface in self.interfaces:
+            gamma = interface.penalty_term
+
+            subdomain_1, subdomain_2 = interface.subdomains
+            b_res, t_res = interface.restriction
+            n_b = n(b_res)
+            n_t = n(t_res)
+            h_b = 2 * cr(b_res)
+            h_t = 2 * cr(t_res)
+
+            all_mobile_species = [spe for spe in self.species if spe.mobile]
+            if len(all_mobile_species) > 1:
+                raise NotImplementedError("Multiple mobile species not implemented")
+            H = all_mobile_species[0]
+
+            v_b = H.subdomain_to_test_function[subdomain_1](b_res)
+            v_t = H.subdomain_to_test_function[subdomain_2](t_res)
+
+            u_b = H.subdomain_to_solution[subdomain_1](b_res)
+            u_t = H.subdomain_to_solution[subdomain_2](t_res)
+
+            K_b = subdomain_1.material.get_solubility_coefficient(
+                self.mesh.mesh, self.temperature_fenics(b_res), H
+            )
+            K_t = subdomain_2.material.get_solubility_coefficient(
```

Comment:
Add reference to DG problem and document derivation of this particular way of writing it.